### ISSUE TITLE
[Request] New log system

### ISSUE THREAD

**Issue by jonisc** [SoftEtherVPN/SoftEtherVPN#260]
Would it be possible, that the logs have a new system like this:

The log of today is simply named eg sec.log. At a specific time (Midnight) Soft Ether creates a folder (OLD) and moves the log into OLD and renames it to sec_<DATE>. Then log reading programs like fail2ban have a easier way to read the logs and as admin you can delete the logs after a time, your Disk space is not too full,… it would be nice too if SoftEther deletes the log after a week / month /... What do you think?
Link to SoftEther User Forum: http://www.vpnusers.com/viewtopic.php?f=7&t=6398&p=17575#p17575

**Comment by benrubson** [comment]
I agree, now that #1122 has been merged, would really be nice for the current security log to have a fixed name, `sec.log`. And to use the timestamped suffix for the rotated files only.
It would help to properly use Fail2Ban.
Thank you 👍

**Comment by chipitsine** [comment]
yeah.... log system revamp
great!

**Comment by benrubson** [comment]
A fixed `sec.log` name would be enough for now, whole revamp is not needed :)

**Comment by chipitsine** [comment]
there's big demand on dealing with IP addresses in various ways. 

it's great you did those changes.

I also think, it might be good idea to document most popular user scenarios, like ip + geo, fail2ban, what do you think ?

**Comment by benrubson** [comment]
Glad to help / improve :)
Of course I can document fail2ban configuration yes.
Could you point me to the correct location / path where documentation resides ?

Do you think you'll be able to handle the fixed `sec.log` modification as proposed [above](https://github.com/SoftEtherVPN/SoftEtherVPN/issues/260#issuecomment-628551736) ?
Sounds like we may have to deal with `LogThreadWriteGeneral ` function, so that current log remains `sec.log`, and old one is renamed to `MakeLogFileName` call result. Something like this...

**Comment by benrubson** [comment]
> Do you think you'll be able to handle the fixed `sec.log` modification as proposed [above](https://github.com/SoftEtherVPN/SoftEtherVPN/issues/260#issuecomment-628551736) ?
> Sounds like we may have to deal with `LogThreadWriteGeneral ` function, so that current log remains `sec.log`, and old one is renamed to `MakeLogFileName` call result. Something like this...

Another solution could be to trap `SIGUSR` (for example), and to re-open all log files when caught.
This would allow to properly rotate log files using an external program (logrotate), we would then be able to disable the own SoftEtherVPN log rotation, leading to fixed-name log files.

Perhaps more work than the first solution though...

**Comment by benrubson** [comment]
@chipitsine any news on this perhaps ?
Many thanks 👍

**Comment by benrubson** [comment]
@chipitsine may I perhaps ask you some news regarding this ?
A fixed `sec.log` would really help using Fail2Ban, thus increasing SoftEtherVPN security level...
Many thanks again 👍

**Comment by benrubson** [comment]
A possible workaround is to use the following security log setting :

![image](https://user-images.githubusercontent.com/6764151/96448776-2e474280-1214-11eb-8838-742fbf65cfda.png)

Security log will then remain `sec.log` (until >2GB, as stated [above](https://github.com/SoftEtherVPN/SoftEtherVPN/issues/260#issuecomment-628551736)).
External log rotation could be done, but SoftEtherVPN then has to be reloaded to properly close / reopen log files.
